infra/src/test/kotlin/fi/fta/geoviite/infra/TestDBService.kt

```diff
@@ -507,6 +508,7 @@ data class TestLayoutContext(
                 .map { (jointNumber, position) ->
                     TrackLayoutSwitchJoint(number = jointNumber, location = position, locationAccuracy = null)
                 },
+            stateCategory = EXISTING,
```

Comment:
Todella monet testit tosiaan hajosivat sen takia, että jos vaihteen testigeneraattorille ei kiinnitä tilakategoriaa, niin se arpoo sieltä fiksatun seedin perusteella Jonkun^TM, ja nyt on suurempi mahdollisuus siihen, että se jokin on `NOT_EXISTING`. Täten tällä tiketillä on lisäilty todella moneen paikkaan näitä kiinnityksiä.

Näistä meidän testigeneraattoreista voisi jutella vähän paremmin vaikka maanantaina dailyssä - minä ja Markus ollaan tämänpäiväisten keskustelujen perusteella vissiin molemmat eletty aiemmin maailmassa jossa testigeneraattorit oikeasti randomisoivat mahdollisimman paljon kaikesta ei-kiinnitetystä datasta, ja ollaan vissiin molemmat aikanamme tykätty siitä.

Reply:
Joo, vähän mielipidekysymys että halutaanko satunnaistaa testidataa (flakyt testit mahdollisempia). Mutta jos jonkin asteinen satunnaistaminen on kuitenkin käytössä, niin silloin testit pitäisi rakentaa niin etteivät ne menisi ei-kiinnittämättömästä datasta rikki ja vastaavasti myös satunnaistamisen pitäisi sitä dataa oikeasti satunnaistaa eri testisuoritusten välillä. Ilmeisesti meillä ei ole satunnaisseedi tai vastaava nyt vaihtunut missään välissä, leading to above 🤔 

Toki meillä voisi olla erilliset funktiot satunnaistetulle sekä kiinnitetyille testidata-asseteille, esim. nykyinen switch() olisi se kiinnitetty, staattinen versio ja vaikkapa randomSwitch() sitten se satunnaistetumpi.

Lisäksi vaihtoehtona on tietysti parametrisoidut testit, joissa se sama testi ajetaan joka kerta useammalla kuin yhdellä alustustilanteella läpi, jolloin flakytestejä ei yhtä herkästi ainakaan synny. Tästä tosin seuraa usein hieman turhaa testien ajamista uudelleen ja uudelleen, joka vie aikaa esim. CI-putkessa (kummallakin [satunnaistettu, parametrisoitu] siis omat käyttötapauksensa minusta)